### DISCUSSION TITLE
Portable Version?

### DISCUSSION THREAD

**Discussion by thejhobe** [Devolutions/UniGetUI#622] [General]
As many of us, I work in a field where "installing" a package manager can be a bit much to do on every server/client. Instead I'm wondering if it's possible to bring this to a self containable portable version? This way I can create lists for what I need to install, fire up the installer (with admin permissions of course) run the install and go. We don't really need it to run. Honestly I installed it on one of my home PC's and had to disable the startup because it kept reminding me of package updates multiple times a day. lol I'd hate for that to happen to a customer.

**Comment by Spacellary** [comment]
_I had a laugh with the thought of the average users finally keeping their software up-to-date because of Winget UI nagging them about it lol People really don't like to update things sometimes. I'd honestly introduce them to Winget UI and encourage them to use it to install things instead of going to websites for .EXEs too xD_

But I understand the use case here, if you're on the business of setting up and shipping PCs to costumers a tool like Winget UI is nice to just install things that aren't already in the usual list your ship, like maybe the user asks for some audio editing and streaming utilities, and while we can always use the CLI for the bulk of the packages you install normally for most people (I guess just have a command ready with everything) it's good to also be able to have a nice GUI to install piece meal software.

**On that same note it would be interesting to have Winget UI generate "install list scripts", as in you can right click a package and "Send it to install Bucket", then do that for all you need and have a button to view the Bucket list, and then generate a install script/command and to also Export and Import those lists. You get the best of both worlds.** (Putting this out here and I'd like to know if this feature would be worth of a request for implementation (if it's not a duplicate or unplanned ofc)

+1

**Reply by marticliment** [reply]
This "export to bulk install" is in the todo list, and it will come, i promise ;)

**Reply by thejhobe** [reply]
You are a treasure. <3

**Reply by Spacellary** [reply]
> This "export to bulk install" is in the todo list, and it will come, i promise ;)

You're an angel Marti <3

**Comment by thejhobe** [comment]
LOL I get that. We aren't shipping/building, we're an MSP and when I commission new hardware there's a small list of some stuff we put on there like web browsers, PDF readers, media player, things that we know are standards and not the baked in stuff MS puts (and changes frequently) in their OS. Or maybe we have a new or complete upgrade and that one client wants all PCs to have xxx Program. 

I've been bouncing between portable app platforms and bootable drives because I hate having to rely on installing everything especially on a server. This would be amazing, I would even say an annual license for this wouldn't be off the table either. 

Now having something that would build a standalone package installer would be amazing. Other than this, we have no reason to install python on every server/workstation. In fact I'd rather not? Plus it defeats the purpose of having a package utility that you just feed it a list and go.

**Reply by marticliment** [reply]
You can do two things to run wingetui as a portable app:

The first one, easy, without further complication.
Run the installer and select a custom folder/drive/usb thumb drive and install it there.
You can run the WingetUI.exe file from the wingetui folder without issues. Winget is bundled also, so you shouldn't have any issues with it.

The second one, more tricky:
Modify the Win.spec file according to pyinstaller documentations and enable onefile mode. Then compile WingetUI from source.

**Reply by thejhobe** [reply]
The first part sounds grand and super easy. The 2nd is a bit over my paygrade lol. But I can try to consult with someone later. Either way, I'll see what I can do because this would be amazing! At this point I'd even settle for the CLI to be portable. I just cringe at the thought of telling my boss "Hey, we need to install Python on servers. Oh and those 2012R2 and 2016 instances yeah that too and I need to do some reg hacks... " lol

**Reply by marticliment** [reply]
what I'm. not sure is about winget compatibility with 2012 servers. 2016 should be w10 based, so there shouldn't be any issues. but with w8 based ones, I'm not really sure if it would work... not becauer of WingetUI, but becauer of winget itself.

**Reply by thejhobe** [reply]
Oh, I'm aware. Fortunately we're phasing out 2012 servers and Win8 machines. So it's not a big deal for me. But when I tried to get WinGet on any 2016 machine there were too many hoops to jump through. Between VS and Python (I don't code... yet) it was more than I was willing to put time into just to save some time to install some apps on them. Now that our base server OS is 2019 and all our workstations are Win10 21h2 or higher, I'm good to go with this.

**Reply by marticliment** [reply]
yes, with 2016/2019  and higher windows server you should be fine.

if there happens to be any issue with the "portable"  install or with windows server compatibility, make sure to open a new support ticket or to reply here, so I can track down and (hopefully) fix the issue!

**Comment by leminsc8er** [comment]
it would be nice if you add "install portable" in your installer without any regedit or autostart erntrys. this is the reasen i don't want install your sofrtware...

**Reply by marticliment** [reply]
You can do that with the latest version...

**Comment by Meche697** [comment]
First time user Ver 1.6.3482. I have the Winget BUG on a portable install. Everything but the package discovery appears to work as I updated packages and can see the installed. As I installed "portable" I'm not sure the work around is supposed to work as it should delete the entry. Anyway, NEW Win 11 in ADMIN account. 

1. Quit WingetUI *DONE*
2. Delete the C:\Users\ADMIN\AppData\Local\Microsoft\WinGet\Settings\defaultState folder and its contents. You may be requested for administrator permission. Click "Yes".    DO THIS FOR EVERY USER ON YOUR MACHINE. *Do we need to do this on Default and Public {Users} too?*
3. Started WingetUI again and disable System Winget if you had enabled it.  *Where do I find "disable System Winget"? I looked everywhere?*

Thank you for the software

**Reply by marticliment** [reply]
> 2\. Do we need to do this on Default and Public {Users} too?

Yes

> 3\. disable System Winget

Under experimental settings

**Comment by vdias** [comment]
Just downloaded the .zip release looking for a portable version, and not quite sure this is portable...

Seeing movement on C:\Users\USER\AppData\Local\UniGetUI.

is there any portable "setup" option for this?
